### PR TITLE
NAS-120537 / 23.10 / If S.M.A.R.T. test status is ABORTED, do not treat it as a pool health issue

### PR DESCRIPTION
**Pull request by bvasilenko**
**Testing**

Run SMART test for a disk. Abort the test by rebooting, while the test is still running

Most probably VM box won't allow to run SMART test, so please use mockup: simulate an aborted test putting `'ABORTED'` into the `status` field.

**Expected Result:**

A warning icon should not be displayed on **Storage** page at **Disk Health** block

### CODE REVIEW

File: src/app/pages/storage/components/dashboard-pool/disk-health-card/disk-health-card.component.ts [hunk 1]
```diff
@@ -76,7 +76,7 @@ export class DiskHealthCardComponent implements OnInit, OnChanges {
   }
 
   get iconType(): PoolCardIconType {
-    if (this.diskState.alerts || this.diskState.smartTests) {
+    if (this.diskState.smartTests) {
```
Comment (dszidi) [inline src/app/pages/storage/components/dashboard-pool/disk-health-card/disk-health-card.component.ts:79]: By suppressing the icon for all alerts, it also won't show when there are temperature related alerts.
We should probably be trying to filter the test status instead

**Review by dszidi — CHANGES_REQUESTED** [review]
See comment above

File: src/app/pages/storage/components/dashboard-pool/disk-health-card/disk-health-card.component.ts (hunk 1 above)
Comment (bvasilenko) [reply src/app/pages/storage/components/dashboard-pool/disk-health-card/disk-health-card.component.ts:79]: I totally agree, thanks for noticing that

**Comment by bvasilenko** [comment]
@dszidi WebUI code will work fine without any change to it's logic. The `ABORTED` statuses will be filtered out by an existing `.filter` call at `getProcessedDisks` method.

**Comment by undsoft** [comment]
@bvasilenko So why was this issue reported?

**Comment by bvasilenko** [comment]
> @bvasilenko So why was this issue reported?

The issue was reported [here](https://ixsystems.atlassian.net/browse/NAS-120498), and fixed [here](https://github.com/truenas/middleware/pull/10764) in the middleware. I don't believe the issue persisted after this PR was merged.

**Pull request merged**